ocaml/bytecomp/bytegen.ml

```diff
@@ -1160,6 +1178,40 @@ and comp_expr_list stack_info env exprl sz cont = match exprl with
       comp_expr stack_info env exp sz
         (Kpush :: comp_expr_list stack_info env rem (sz+1) cont)
 
+(* Compile a list of reuse arguments [x; e1; ...; eN] and their resets
+   [None; Some _; ...] to a primitive operation. For each None, we push
+   a read on x on the stack and for each Some _, we push an ei on the stack.
+   As with regular arguments, we push in reversed order and
+   leave the first argument in the accumulator. *)
+
+and comp_reuse_args stack_info env argl resets getfield sz cont =
+  let rec zip_resets idx resets argl =
```

Comment:
This could be made tail-recursive -- and it would remove the need for the `List.rev` after it's done.